File: src/main/java/de/ayont/lpc/renderer/SpigotChatRenderer.java
package de.ayont.lpc.renderer;

import de.ayont.lpc.LPC;
import net.kyori.adventure.audience.Audience;
import net.kyori.adventure.text.Component;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class SpigotChatRenderer {
    private final LPC plugin;

    public SpigotChatRenderer(LPC plugin) {
        this.plugin = plugin;
    }

    public @NotNull Component render(Player source, String message, Audience viewer) {
        return plugin.getChatRendererUtil().render(source, message, viewer, null);
    }
}

File: src/main/kotlin/de/ayont/lpc/renderer/SpigotChatRenderer.kt
package de.ayont.lpc.renderer

import de.ayont.lpc.LPC
import net.kyori.adventure.audience.Audience
import net.kyori.adventure.text.Component
import org.bukkit.entity.Player

class SpigotChatRenderer(private val plugin: LPC) {
    fun render(source: Player, message: String, viewer: Audience): Component {
        return plugin.chatRendererUtil?.render(source, message, viewer, null) ?: Component.empty()
    }
}
